File: Csharp/Creational/Factory/FY4/Factory/Director.cs
namespace TheRayCode.Factory.DesignPatterns
{
    public class Director
    {
        private Builder _builder;
        
        public Builder Builder
        {
            set { _builder = value; } 
        }
        
        // The Director can construct several product variations using the same
        // building steps.
        public void buildMinimalViableProduct()
        {
            this._builder.BuildPartA();
        }
        
        public void buildFullFeaturedProduct()
        {
            this._builder.BuildPartA();
            this._builder.BuildPartB();
            this._builder.BuildPartC();
        }
    }


}

File: PHP/Creational/Builder/Example/Director.php
<?php
include_once 'Builder.php';
class Director
{
    private Builder $builder;

    // Set the builder to use
    public function setBuilder(Builder $builder): void
    {
        $this->builder = $builder;
    }

    // Construct the product step by step
    public function construct(): void
    {
        $this->builder->buildPartA();
        $this->builder->buildPartB();
    }

}